### PR TITLE
feat: add compatibility to kubernetes 1.28-1.29

### PR DESCRIPTION
**Pull request by sbruzzese902**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -198,77 +195,58 @@ trigger:
...
-  - name: init
-    image: quay.io/sighup/e2e-testing-drone-plugin:v1.26.4
+  - name: create-kind-cluster
+    image: quay.io/sighup/dind-kind-kubectl-kustomize:0.20.0_1.29.1_3.10.0
     pull: always
     volumes:
+      - name: dockersock
+        path: /var/run/docker.sock
       - name: shared
         path: /shared
```
Comment (ralgozino) [inline .drone.yml:205]: you don't need the shared volume. The `workdir` (`/drone/src`) where the repo is cloned is shared between the pipelines
```suggestion
```

File: .drone.yml [hunk 2]
```diff
@@ -198,77 +195,58 @@ trigger:
...
 
   - name: e2e
-    # KUBECTL 1.26.3 - KUSTOMIZE 3.5.3 - HELM 3.1.1 - YQ 4.21.1 - ISTIOCTL 1.9.4 - FURYCTL 0.9.0 - BATS 1.1.0
+    # KUBECTL 1.27.1 - KUSTOMIZE 3.5.3 - HELM 3.1.1 - YQ 4.21.1 - ISTIOCTL 1.9.4 - FURYCTL 0.9.0 - BATS 1.1.0
     image: quay.io/sighup/e2e-testing:1.1.0_0.11.0_3.1.1_1.9.4_1.26.3_3.5.3_4.33.3
     pull: always
+    network_mode: host
     volumes:
       - name: shared
         path: /shared
```
Comment (ralgozino) [inline .drone.yml:228]:
```suggestion
```

File: .drone.yml [hunk 3]
```diff
@@ -277,17 +255,18 @@ steps:
 volumes:
   - name: shared
     temp: {}
```
Comment (ralgozino) [inline .drone.yml:257]:
```suggestion
```

File: .drone.yml [hunk 4]
```diff
@@ -198,77 +195,58 @@ trigger:
...
-        from_secret: dockerhub_username
-      dockerhub_password:
-        from_secret: dockerhub_password
+    environment:
+      CLUSTER_VERSION: v1.27.3
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
+      # /drone/src is the default workdir for the pipeline
+      # using this folder we don't need to mount another
+      # shared volume between the steps
+      KUBECONFIG: /drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:213]: Add some unique identifier to the kubeconfig path, otherwise it will be overwritten by the other pipelines that run in parallel for the other kubernetes versions.
```suggestion
      KUBECONFIG: /drone/src/kubeconfig-127
```

File: .drone.yml [hunk 5]
```diff
@@ -198,77 +195,58 @@ trigger:
...
     pull: always
+    network_mode: host
     volumes:
       - name: shared
         path: /shared
-    depends_on: 
-    - init
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
+      KUBECONFIG: /drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:231]:
```suggestion
      KUBECONFIG: /drone/src/kubeconfig-127
```

File: .drone.yml [hunk 6]
```diff
@@ -277,17 +255,18 @@ steps:
...
+name: e2e-kubernetes-1.28
 kind: pipeline
 type: docker
 
 depends_on:
   - policeman
-
-node:
-  runner: internal
+  - e2e-kubernetes-1.27
```
Comment (ralgozino) [inline .drone.yml:269]: remove the dependency so we run the e2e's in parallel :)
```suggestion
```

File: .drone.yml [hunk 7]
```diff
@@ -198,77 +195,58 @@ trigger:
...
     image: quay.io/sighup/e2e-testing:1.1.0_0.11.0_3.1.1_1.9.4_1.26.3_3.5.3_4.33.3
     pull: always
+    network_mode: host
     volumes:
       - name: shared
         path: /shared
-    depends_on: 
-    - init
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
```
Comment (ralgozino) [inline .drone.yml:230]:
```suggestion
      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-127
```

File: .drone.yml [hunk 8]
```diff
@@ -176,16 +176,13 @@ steps:
         - tag
 
 ---
-name: e2e-kubernetes-1.26
+name: e2e-kubernetes-1.27
 kind: pipeline
 type: docker
```
Comment (ralgozino) [inline .drone.yml:182]: You add a directive to clone the last commit instead of the full repo history:
```suggestion
clone:
  depth: 1
```

File: .drone.yml [hunk 9]
```diff
@@ -301,77 +280,58 @@ trigger:
...
-        from_secret: dockerhub_username
-      dockerhub_password:
-        from_secret: dockerhub_password
+    environment:
+      CLUSTER_VERSION: v1.28.0
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
+      # /drone/src is the default workdir for the pipeline
+      # using this folder we don't need to mount another
+      # shared volume between the steps
+      KUBECONFIG: /drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:298]:
```suggestion
      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-128
      # /drone/src is the default workdir for the pipeline
      # using this folder we don't need to mount another
      # shared volume between the steps
      KUBECONFIG: /drone/src/kubeconfig-128
```

File: .drone.yml [hunk 10]
```diff
@@ -301,77 +280,58 @@ trigger:
...
     pull: always
+    network_mode: host
     volumes:
       - name: shared
         path: /shared
-    depends_on: 
-    - init
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
+      KUBECONFIG: /drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:316]:
```suggestion
      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-128
      KUBECONFIG: /drone/src/kubeconfig-128
```

File: .drone.yml [hunk 11]
```diff
@@ -301,77 +280,58 @@ trigger:
...
-      vsphere_password:
-        from_secret: vsphere_password
-      vsphere_user:
-        from_secret: vsphere_user
-      dockerhub_username:
-        from_secret: dockerhub_username
-      dockerhub_password:
-        from_secret: dockerhub_password
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
```
Comment (ralgozino) [inline .drone.yml:331]:
```suggestion
      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-128
```

File: .drone.yml [hunk 12]
```diff
@@ -380,16 +340,104 @@ steps:
 volumes:
   - name: shared
     temp: {}
+  - name: dockersock
+    host:
+      path: /var/run/docker.sock
```
Comment (ralgozino) [inline .drone.yml:345]:
```suggestion
  - name: dockersock
    host:
      path: /var/run/docker.sock
```

File: .drone.yml [hunk 13]
```diff
@@ -380,16 +340,104 @@ steps:
...
+
+---
+name: e2e-kubernetes-1.29
+kind: pipeline
+type: docker
+
```
Comment (ralgozino) [inline .drone.yml:351]:
```suggestion
clone:
  depth: 1

```

File: .drone.yml [hunk 14]
```diff
@@ -380,16 +340,104 @@ steps:
...
+depends_on:
+  - policeman
+  - e2e-kubernetes-1.28
```
Comment (ralgozino) [inline .drone.yml:354]:
```suggestion
```

File: .drone.yml [hunk 15]
```diff
@@ -380,16 +340,104 @@ steps:
...
+
+steps:
+  - name: create-kind-cluster
+    image: quay.io/sighup/dind-kind-kubectl-kustomize:0.20.0_1.29.1_3.10.0
+    pull: always
+    volumes:
+      - name: dockersock
+        path: /var/run/docker.sock
+      - name: shared
+        path: /shared
```
Comment (ralgozino) [inline .drone.yml:375]:
```suggestion
    volumes:
      - name: dockersock
        path: /var/run/docker.sock
```

File: .drone.yml [hunk 16]
```diff
@@ -380,16 +340,104 @@ steps:
...
+    depends_on: [clone]
+    environment:
+      CLUSTER_VERSION: v1.29.0
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
+      # /drone/src is the default workdir for the pipeline
+      # using this folder we don't need to mount another
+      # shared volume between the steps
+      KUBECONFIG: /drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:383]:
```suggestion
      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-129
      # /drone/src is the default workdir for the pipeline
      # using this folder we don't need to mount another
      # shared volume between the steps
      KUBECONFIG: /drone/src/kubeconfig-129
```

File: .drone.yml [hunk 17]
```diff
@@ -380,16 +340,104 @@ steps:
...
+      - kind get kubeconfig --name $${CLUSTER_NAME} > $${KUBECONFIG}
+
+  - name: e2e
+    # KUBECTL 1.29.1 - KUSTOMIZE 3.5.3 - HELM 3.1.1 - YQ 4.21.1 - ISTIOCTL 1.9.4 - FURYCTL 0.9.0 - BATS 1.1.0
+    image: quay.io/sighup/e2e-testing:1.1.0_0.11.0_3.12.0_1.9.4_1.29.1_3.5.3_4.33.3
+    pull: always
+    network_mode: host
+    volumes:
+      - name: shared
+        path: /shared
```
Comment (ralgozino) [inline .drone.yml:398]:
```suggestion
```

File: .drone.yml [hunk 18]
```diff
@@ -380,16 +340,104 @@ steps:
...
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
+      KUBECONFIG: /drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:401]:
```suggestion
      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-129
      KUBECONFIG: /drone/src/kubeconfig-129
```

File: .drone.yml [hunk 19]
```diff
@@ -380,16 +340,104 @@ steps:
...
+    depends_on: [create-kind-cluster]
+    commands:
+      - export KUBECONFIG=/drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:404]: You are already exporting the path in the `environment` section above.
```suggestion
```

File: .drone.yml [hunk 20]
```diff
@@ -198,77 +195,58 @@ trigger:
...
         path: /shared
-    depends_on: 
-    - init
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
+      KUBECONFIG: /drone/src/kubeconfig
+    depends_on: [create-kind-cluster]
     commands:
-      - export KUBECONFIG=/shared/kube/kubeconfig-126
+      - export KUBECONFIG=/drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:234]: You are already exporting the env var in the `environment` section above
```suggestion
```

File: .drone.yml [hunk 21]
```diff
@@ -301,77 +280,58 @@ trigger:
...
         path: /shared
-    depends_on: 
-    - init
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
+      KUBECONFIG: /drone/src/kubeconfig
+    depends_on: [create-kind-cluster]
     commands:
-      - export KUBECONFIG=/shared/kube/kubeconfig-127
+      - export KUBECONFIG=/drone/src/kubeconfig
```
Comment (ralgozino) [inline .drone.yml:319]:
```suggestion
```

File: .drone.yml [hunk 22]
```diff
@@ -380,16 +340,104 @@ steps:
...
+
+  - name: delete-kind-cluster
+    image: quay.io/sighup/dind-kind-kubectl-kustomize:0.20.0_1.29.1_3.10.0
+    volumes:
+      - name: dockersock
+        path: /var/run/docker.sock
+    pull: always
+    depends_on: [e2e]
+    environment:
+      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}
```
Comment (ralgozino) [inline .drone.yml:416]:
```suggestion
      CLUSTER_NAME: ${DRONE_REPO_NAME}-${DRONE_BUILD_NUMBER}-129
```

File: .drone.yml [hunk 23]
```diff
@@ -380,16 +340,104 @@ steps:
...
+      status:
+        - success
+        - failure
+
+volumes:
+  - name: shared
+    temp: {}
+  - name: dockersock
+    host:
+      path: /var/run/docker.sock
```
Comment (ralgozino) [inline .drone.yml:430]:
```suggestion
volumes:
  - name: dockersock
    host:
      path: /var/run/docker.sock
```

File: README.md [hunk 24]
```diff
@@ -5,7 +5,7 @@
 </h1>
 <!-- markdownlint-enable MD033 -->
 
-![Release](https://img.shields.io/badge/Latest%20Release-v3.0.2-blue)
+![Release](https://img.shields.io/badge/Latest%20Release-v3.0.3-blue)
```
Comment (ralgozino) [inline README.md:8]: This should be a 3.1.0 instead of a 3.0.3; we are bumping minor versions of the components and dropping support for older kubernetes versions, this is not a bug fix.

Note: I'll comment once here to not spam the PR, the version should be changed in several places.

File: docs/releases/v3.0.3.md [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+| `mimir`               | [`v2.11.0`](https://github.com/grafana/mimir/releases/tag/mimir-2.11.0)                             | `v2.10.3`        |
+| `minio-ha`            | [`RELEASE.2024-02-09T21-25-16Z`](https://github.com/minio/minio/tree/RELEASE.2024-02-09T21-25-16Z)  | `RELEASE.2023-01-12T02-06-16Z` |
+
+> Please refer the individual release notes to get a detailed info on the releases.
+
+## Update Guide 🦮
+
+### Process
+
+To upgrade this core module from `v3.0.1` to `v3.0.2`, execute the following:
```
Comment (ralgozino) [inline docs/releases/v3.0.3.md:32]: this is referencing the previous release

File: katalog/mimir/MAINTENANCE.md [hunk 26]
```diff
@@ -17,6 +17,19 @@ helm template mimir-distributed /tmp/mimir-distributed -n monitoring --values MA
...
+Move the dashboards in place:
+```bash
+cp /tmp/mimir-distributed/mixins/dashboards/*.json dashboards
+```
+
+Move the prometheus rules in place:
+```bash
+helm template mimir-distributed /tmp/mimir-distributed \
+  --set metaMonitoring.prometheusRule.enabled=true --set metaMonitoring.prometheusRule.mimirAlerts=true --set metaMonitoring.prometheusRule.mimirRules=true \
+  -s templates/metamonitoring/mixin-alerts.yaml -s templates/metamonitoring/prometheusrule.yaml -s templates/metamonitoring/recording-rules.yaml \
```
Comment (ralgozino) [inline katalog/mimir/MAINTENANCE.md:29]: why do we enable monitoring componets via a flag and disable them in the values file? what am I missing?

**Review by ralgozino — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 2 above)
Comment (mimnix) [reply .drone.yml]: Solved by 1f6db8b91cbc876fb5776596c919c4151fe42333

File: .drone.yml (hunk 3 above)
Comment (mimnix) [reply .drone.yml]: Solved by 1f6db8b91cbc876fb5776596c919c4151fe42333

File: docs/releases/v3.0.3.md (hunk 25 above)
Comment (mimnix) [reply docs/releases/v3.0.3.md]: Solved by 5fe5cc04a6ff19047cb4591b14ad4f3a5e0149cc

File: katalog/mimir/MAINTENANCE.md (hunk 26 above)
Comment (nutellinoit) [reply katalog/mimir/MAINTENANCE.md:29]: I think it's a way to get pieces of the helm templating with -s, maybe the --set overrides stuff in the values.yaml

File: katalog/mimir/config/mimir.yaml [hunk 27]
```diff
@@ -58,24 +58,22 @@ limits:
   max_cache_freshness: 10m
   max_query_parallelism: 240
   max_total_query_length: 12000h
-  compactor_blocks_retention_period: 720h # 30 days retention by default
```
Comment (nutellinoit) [inline katalog/mimir/config/mimir.yaml:61]: Why was this removed?

**Review by nutellinoit — CHANGES_REQUESTED** [review]
Only one question below 👇

File: katalog/mimir/config/mimir.yaml (hunk 27 above)
Comment (mimnix) [reply katalog/mimir/config/mimir.yaml:61]: Taken from upstream, we should keep the property from now on
